File: articles/risk-analysis-and-probabilistic-reasoning/r/run_all_risk_analysis_workflows.R
# run_all_risk_analysis_workflows.R
# Run all R workflows for Risk Analysis and Probabilistic Reasoning.

args <- commandArgs(trailingOnly = FALSE)
file_arg <- grep("^--file=", args, value = TRUE)

if (length(file_arg) > 0) {
  script_path <- normalizePath(sub("^--file=", "", file_arg[1]), mustWork = TRUE)
  article_root <- normalizePath(file.path(dirname(script_path), ".."), mustWork = TRUE)
} else {
  article_root <- getwd()
}

r_dir <- file.path(article_root, "r")

scripts <- c(
  "risk_analysis_probabilistic_reasoning_workflow.R",
  "expected_loss_profiles.R",
  "tail_exposure_report.R",
  "stress_test_tables.R",
  "probability_quality_summary.R",
  "bayesian_update_tables.R"
)

for (script in scripts) {
  path <- file.path(r_dir, script)
  message("Running ", path)
  source(path)
}

message("All R risk-analysis workflows completed.")

File: articles/risk-analysis-and-probabilistic-reasoning/python/run_all_risk_analysis_workflows.py
#!/usr/bin/env python3
"""Run all Python workflows for Risk Analysis and Probabilistic Reasoning."""

from __future__ import annotations

from pathlib import Path
import subprocess
import sys

ARTICLE_ROOT = Path(__file__).resolve().parents[1]
PYTHON_DIR = ARTICLE_ROOT / "python"

SCRIPTS = [
    "expected_loss_calculator.py",
    "variance_and_tail_metrics.py",
    "value_at_risk_cvar.py",
    "threshold_breach_analysis.py",
    "scenario_stress_testing.py",
    "probability_quality_audit.py",
    "bayesian_risk_update.py",
    "decision_record_exporter.py",
    "risk_analysis_probabilistic_reasoning_simulation.py",
]


def main() -> None:
    for script in SCRIPTS:
        path = PYTHON_DIR / script
        print(f"Running {path.name}...")
        subprocess.run([sys.executable, str(path)], check=True)
    print("All Python risk-analysis workflows completed.")


if __name__ == "__main__":
    main()
